tests/sentry/web/frontend/test_csrf.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+from importlib import import_module
+
+from django.conf import settings
+from django.core.urlresolvers import reverse, NoReverseMatch
+
+from sentry.testutils import TestCase
+
+
+class CSRFTest(TestCase):
+    def test_endpoints_have_csrf(self):
+        """
+        Iterate on all endpoints that have unambiguous paths and send a POST
+        request with no CSRF token (either in params or cookie) and validate
+        that the response is a 403 with the CSRF warning.
+        """
+        user = self.create_user('foo@example.com')
+        self.login_as(user)
+        self.client.handler.enforce_csrf_checks = True
+
+        protected = {True: 0, False: 0}
+        urlconf = import_module(settings.ROOT_URLCONF)
+        patterns = [u for u in urlconf.urlpatterns if hasattr(u, 'name') and u.name is not None]
+        for p in patterns:
+
+            # need to delete this cookie every time
+            if settings.CSRF_COOKIE_NAME in self.client.cookies:
+                del self.client.cookies[settings.CSRF_COOKIE_NAME]
+
+            try:
+                path = reverse(p.name)
+            except NoReverseMatch:
+                continue
+            try:
+                resp = self.client.post(path)
+                is_protected = (
+                    resp.status_code == 403 and
+                    "A required security token was not found or was invalid." in resp.content
+                )
+                protected[is_protected] += 1
```

Comment:
surely all this tests is that one endpoint has csrf?

Reply:
"Obviously some endpoints are deliberately exempt, so the question of what to assert at the end remains open"  :)

Reply:
I like @dcramer 's idea of making this a base test that any of the view tests can opt in to, or maybe actually opt-out of) Will try come up with something along those lines.

Reply:
I would propose we attach metadata to exempt endpoints and then assert on the rest. Also we can enforce CSRF for session auth only.

Reply:
yeah, the problem is that basically the metadata is already there in the form of the `@csrf_protect/csrf_exempt` decorator. but this test is supposed to guard against the case where we accidentally marked all tests as exempt (#6320)

I'm less convinced that this is going to be a useful test now, in the sense that it would either be:
"does this code do what it says it's going to do?" in which case the answer is always yes
or
"does this code do the Right Thing™ for each endpoing, despite what the code says?" in which case we need to keep an independent record of what the Right Thing™ is for each endpoint.